test/elixir_analyzer/exercise_test/feature/block_ends_with_test.exs

```diff
@@ -0,0 +1,477 @@
+defmodule ElixirAnalyzer.ExerciseTest.Feature.BlockEndsWithTest do
+  use ElixirAnalyzer.ExerciseTestCase,
+    exercise_test_module: ElixirAnalyzer.Support.AnalyzerVerification.Feature.BlockEndsWith
+
+  describe "finding a single expression" do
+    test_exercise_analysis "code with an :ok in it",
+      comments_exclude: ["cannot detect :ok"] do
+      [
+        defmodule MyModule do
+          def foo() do
+            :ok
+          end
+        end,
+        defmodule MyModule do
+          def foo() do
+            x = "hi"
+            :ok
+          end
+        end,
+        defmodule MyModule do
+          def foo(:ok) do
+            IO.puts("hello")
+          end
+        end,
+        defmodule MyModule do
+          def foo() do
+            case {} do
+              _ -> :ok
+            end
+          end
+        end,
+        defmodule MyModule do
+          def foo() when :ok do
+            true
+          end
+        end,
+        defmodule MyModule do
+          def foo() do
+            :ok.lets_pretend_this_is_an_erlang_module()
+            # Erlang function call counts as atom literal
+          end
+        end,
+        defmodule MyModule do
+          def foo() do
+            :ok
+            # Atoms are always found, even when they are not the last
+            :nope
+          end
```

Comment:
What's going on here? Why does it find the `:ok` atom even though it's not last?

Reply:
Maybe the comment is not accurate. With this form:

```elixir 
form do
  _block_ends_with do
    :ok
  end
end
```
`_block_ends_with` will get compared to every single node in the code AST, not just groups of node, so it will match any `:ok` anywhere (because a single node is also technically the last).

However here
```elixir 
form do
  def foo() do
    _block_ends_with do
      :ok
    end
  end
end
```
`_block_ends_with` will only get compared to the group of lines inside of `foo`, so it will only match an `:ok` that is at the last line. I've added some tests that make this behavior ore clear.

I know this is confusing, but I can't really think of a way out here, other than writing clear documentation.

Reply:
Ah, I get it now. The updated tests are enough, thanks 🙂